lib/issuer.js

```diff
@@ -188,33 +230,206 @@ api.claim = function(actor, id, callback) {
-        fields: {issuer: 1, 'credential.sysState': 1, id: 1},
-        options: {unique: true, background: false}
-      }], callback);
+// TODO: more acceptable formats?
+function checkHeaders(req, res, next) {
+  var acceptableContentTypes = [
+    'application/json',
+    'application/ld+json',
+    'text/html'
+  ];
+  if(!req.accepts(acceptableContentTypes)) {
+    res.status(406);
+    res.write(JSON.stringify({
+      message: 'Requested content types not acceptable.',
+      details: {
+        accepted: req.get('accept'),
+        acceptable: acceptableContentTypes
+      }
+    }));
+    res.end();
+    return next();
+  }
+  if(acceptableContentTypes.indexOf(req.get('Content-Type')) == -1) {
+    res.status(415).end();
+    return next();
+  }
+  next();
+};
+
+function findCredentialByClaimId(claimId, callback) {
+  console.log('CLAIMID:', claimId);
+  query = {
+    'credential.claim.id': claimId
+  };
+  store.provider.getAll(null, query, callback);
+};
+
+bedrock.events.on('bedrock-express.configure.routes', function(app) {
+
+  // parse application/x-www-form-urlencoded
+  var parseForm = bodyParser.urlencoded({extended: false});
+
+  // parse JSON in form data
+  function parseJsonData(req, res, next) {
+    try {
+      req.body = JSON.parse(req.body.jsonPostData);
+    } catch(e) {
+      return next(new BedrockError('Invalid JSON.', 'JSONParseError', {
+        httpStatusCode: 400,
+        public: true
+      }));
     }
-  }, function(err) {
-    callback(err);
+    return next();
+  }
+
+  // FIXME: determine proper endpoint
+  app.get('/creds/claimid/:claimid', function(req, res, next) {
+    var claimId = req.params.claimid;
+    findCredentialByClaimId(claimId, function(err, results) {
+      if(err) {
+        return next(new BedrockError(
+          'The credential could not be located.',
+          'CredentialLookupFailed', {
+            httpStatusCode: 400,
+            'public': true
+          }, err));
+      }
+      return res.json(results);
+    });
+  });
+  // FIXME: this is a provisional endpoint and authentication is required
+  app.put('/creds/:id', function(req, res, next) {
+    var credentialId = req.params.id;
+    console.log('BODY *****', credentialId, req.body);
+    var now = Date.now();
+    if(req.body.accepted) {
+      var updateQuery = {id: credentialId};
+      var updateParams = {
+        $set: {
+          'meta.updated': now,
+          'meta.acceptance': {
+            status: 'accepted',
+            date: now
+          }
+        }
+      };
+      store.provider.collection.update(
+        updateQuery, updateParams, function(err, result) {
+          if(err) {
+            return next(new BedrockError(
+              'The credential could not be updated.',
+              'UpdateCredentialFailed', {
+                httpStatusCode: 400,
+                'public': true
+              }, err));
+          }
+          return res.json({status: 'OK'});
+        });
+    } else {
```

Comment:
Style/Readbility nits: If you've got a simple else block, invert the if conditional, move it to the top and return early, then you can unindent the larger if block. Readers can see weed out the simpler logic paths more quickly and then get to the more complicated ones last (and there will be less indenting).